universal-application-tool-0.0.1/app/assets/javascripts/azure_upload.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * This is responsible for uploading a file to Azure blob storage.
+ */
+
+class AzureUploadController {
+  static UPLOAD_CONTAINER_ID = 'azure-upload-form-component'
+
+  constructor() {
+    const uploadContainer = document.getElementById(
+      AzureUploadController.UPLOAD_CONTAINER_ID
+    )
+    uploadContainer.addEventListener('submit', (event) =>
+      this.attemptUpload(event, uploadContainer)
+    )
```

Comment:
issue (blocking): with these DOM queries in the constructor this code is a little brittle to its load/initialization pattern i.e. if it were loaded asynchronously or injected into the page before the DOM elements it's looking for it would fail.

possible solution: initialize the class after the `load` event https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event

Reply:
Let me know how the updated code looks to you

Reply:
Discussed offline and fixed!